src/magic/token.rs

```diff
@@ -0,0 +1,141 @@
+use logos::{Lexer, Logos};
+
+#[derive(Logos, Debug, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum Token {
```

Comment:
this is just a rough draft - we'll probably iterate on it. right now it's just a simple imperative lang with Ints, Bools, Structs, Enums